### PR TITLE
Update google-cloud-datastore-database.js

### PR DESCRIPTION
**Pull request by joalafu**
addOrUpdateEntity -> "result[0].mutationResults[0].key.path[0].id" no found remplace by "entity.key.id"
findById -> add return for case Invalid entity Id.
replaceById -> this removes does not update. "deleteData" remplace by "addOrUpdateEntity". 
updateInternal -> data was lost. get oldData, merge with newData and update entity.
update -> I only considered updating by id.
updateInternal & deleteData-> throw not working properly, remplace "throw new Error" -> "return Promise.reject".

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by joalafu** [comment]
ddOrUpdateEntity -> "result[0].mutationResults[0].key.path[0].id" no found remplace by "entity.key.id"
findById -> add return for case Invalid entity Id.
replaceById -> this removes does not update. "deleteData" remplace by "addOrUpdateEntity".
updateInternal -> data was lost. get oldData, merge with newData and update entity.
update -> I only considered updating by id.
updateInternal & deleteData-> throw not working properly, remplace "throw new Error" -> "return Promise.reject".

**Comment by henriquecarv** [comment]
> Changes needed to support multiple updates
Thanks for the heads up! :+1: 

Will take care of that in another PR Solution.
Feel free to implement it if you would like.

File: lib/google-cloud-datastore-database.js [hunk 1]
```diff
@@ -62,7 +62,7 @@ class GoogleCloudDatastore {
 
       if (operation === 'save') {
         result = await this.db.insert(entity);
-        const id = result[0].mutationResults[0].key.path[0].id;
+        const id = entity.key.id;
```
[the remark this answers is not in the document]
Comment (joalafu) [reply lib/google-cloud-datastore-database.js]: Yes, you're right, but "result [0] .mutationResults [0] .key" returned me undefined.
I will check it again

File: lib/google-cloud-datastore-database.js [hunk 2]
```diff
@@ -391,16 +393,14 @@ class GoogleCloudDatastore {
    * @param {Object} data The property/value pairs to be updated
    */
   async updateInternal(model, id, data) {
-    if (!id) throw new Error('Provide a valid entity Id');
+    if (!id) return Promise.reject('Provide a valid entity Id');
```
[the remark this answers is not in the document]
Comment (joalafu) [reply lib/google-cloud-datastore-database.js]: "loopback" uses "promise". This avoid the use of "try" "catch".

**Pull request merged**